app/controllers/application_modal_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ApplicationModalController < ApplicationController
```

Comment:
OK, it probably could be cool to render a page in the both ways in some situations. Please, provide examples of controllers/actions where we really need this to justify bringing all this complexity to controllers and weird-looking code in `prepare_to_respond` and `render` actions which I find really dirty and not needed there. Why do you prefer this way over modifying `link_to` like in this [article](https://coderwall.com/p/ej0mhg/open-a-rails-form-with-twitter-bootstrap-modals)?